engine/classes/Elgg/Database/AccessCollections.php

```diff
@@ -369,6 +371,10 @@ public function getWhereSql(array $options = array()) {
 			}
 			$clauses_str .= '(' . implode(' AND ', $clauses['ands']) . ')';
 		}
+		
+		if (empty($clauses_str)) {
+			$clauses_str = "$table_alias{$options['access_column']} = " . ACCESS_PUBLIC;
```

Comment:
Because I don't really understand what we are fixing. We need tests to see the problem and the fix.